### DISCUSSION TITLE
Have outline view include collapsible list of references to symbol

### DISCUSSION THREAD

**Discussion by pdow** [codefori/vscode-ibmi#1356] [Ideas]
Currently, the outline view for an RPGLE program simply shows a list of symbols from the source code. Clicking on a symbol repositions the source view to the definition of the symbol.  This is very helpful.

To make this even more useful, please add a collapsible list of source statements (under the symbol in the outline view) where the symbol is referenced.  For example, in the HelloWorld program, this is what the outline view currently (v1.9.3) looks like:

![image](https://github.com/halcyon-tech/vscode-ibmi/assets/19736064/2bac4efb-cd7f-40d7-ad8e-2668477a5427)

I'd like to see it show

OUTLINE
   printf INT(10) EXTPROC('PRINTF')
     stmt 12
  mytext CHAR(25)
    stmt 11
    stmt 12

Or something like that.

**Comment by worksofliam** [comment]
We won't be doing this with the outline view, but you can right click on any variable in the source and find the references to it or have it take you to the definition.

**Reply by Gregory1701** [reply]
@worksofliam - I get that you are encouraging the use of the references extension  tab dohickey (whatever it's called lol).  I can get it to work on very simple programs, such as this:
![image](https://github.com/halcyon-tech/vscode-ibmi/assets/132368707/0de1ff07-067b-4d2c-9fad-37a421010a69)
But most programs, I usually get a blank screen, even though I can see it's defined in that same procedure.  I'll start another discussion on that.  

For this request, however, I think it has (a lot of) merit.  Would you give this some consideration please?

**Reply by worksofliam** [reply]
I'm more curious about finding out why it doesn't work.

If you are able to recreate it and share some code so I can recreate it.. perhaps we can improve it.

**Reply by Gregory1701** [reply]
[https://github.com/orgs/halcyon-tech/discussions/1373](url)
